corehq/apps/app_manager/xform.py

```diff
@@ -495,33 +496,43 @@ def update_block(self):
         self.elem.append(update_block)
         return update_block
 
-    def add_case_updates(self, updates, make_relative=False):
+    def add_case_updates(self, updates, make_relative=False, case_id_xpath=None):
+        from corehq.apps.app_manager.models import ConditionalCaseUpdate
         update_block = self.update_block
         if not updates:
             return
 
         update_mapping = {}
         attachments = {}
         for key, value in updates.items():
-            value = getattr(value, "question_path", value)
+            value_str = getattr(value, "question_path", value)
             if key == 'name':
                 key = 'case_name'
-            if self.is_attachment(value):
-                attachments[key] = value
+            if self.is_attachment(value_str):
+                attachments[key] = value_str
             else:
                 update_mapping[key] = value
 
         for key, q_path in sorted(update_mapping.items()):
+            resolved_path = self.xform.resolve_path(q_path)
+            edit_mode_path = ''
+            if isinstance(q_path, ConditionalCaseUpdate) and q_path.update_mode == UPDATE_MODE_EDIT:
+                if 'commcare_usercase' in self.path:
+                    case_value = UsercaseXPath().case().slash(key)
+                else:
+                    case_id_xpath = case_id_xpath if case_id_xpath else session_var('case_id')
```

Comment:
I was thinking about this and I think you should either always pass in the xpath or else you can reference the `case_id` attribute of the case block itself which would prevent you from having to pass in the xpath:

case_id_xpath = f"{self.path}case/@case_id"

Reply:
@snopoke I added optional param that because for the advanced forms I found that the `self.path` is not what we wanted (it would be something like `case_id` when we wanted `case_id_load_1` in the test case example). Does that sound accurate?

Reply:
I think maybe you're misunderstanding me.

`self.path` should be the relative path from the data node to the case block node:

```xml
<data>
  <parent>
    <case @case_id=""></case>
  </parent>
</data>
```

`path = 'parent/'`

If path is an empty string then the case is a direct child of the data node. If you know the path to the case element then you can get the case ID: `{path}/case/@case_id`.

Does that make sense?